### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by diversemix**

### CODE REVIEW

File: src/mock-event-bus/index.ts [hunk 1]
```diff
@@ -1,64 +1,41 @@
...
-    private queues: Option<Map<string, AnyHandler>> = None;
+    private queues: Map<string, AnyHandler> = new Map();
+    private eventsToHandle: EventType[];
+    private serviceName: string;
 
-    // This lint suppression wouldn't be needed if these params were passed into constructor
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    public async init(_defs: string[], _serviceName: string): Promise<this> {
-        this.queues = Some(new Map());
+    public async register(eventsToHandle: EventType[], serviceName: string): Promise<this> {
```
Comment (Cooryd) [inline src/mock-event-bus/index.ts]: What was the decision behind not using `constructor` again? I think there may be a chance of confusion between `register` and `subscribe` methods on this object

File: src/rabbit-event-bus/amqp-connector.ts [hunk 2]
```diff
@@ -161,6 +110,81 @@ export default class AMQPConnector {
...
+                    rabbitChannel.ack(msg);
+                } else {
+                    // Nack
+                    logger.warn('eventHandlerFailure');
+                    rabbitChannel.nack(msg, false, true);
+                }
+            });
+        } catch (e) {
+            rabbitChannel.nack(msg, false, false);
+            logger.warn(`Can't parse JSON! Error: ${e}`);
```
Comment (Cooryd) [inline src/rabbit-event-bus/amqp-connector.ts]: Would it be useful to include any extra output to pinpoint this function as the source of the error? Incase we have other `JSON.parse` calls being try/catched in future?

File: src/rabbit-event-bus/index.ts [hunk 3]
```diff
@@ -20,21 +20,29 @@ export interface RabbitEventBusConnectionOptions {
...
 
+    public get connector(): Option<AMQPConnector> {
+        return this._connector;
+    }
+
+    public get subscriptions(): Array<Subscription<unknown & object>> {
+        return this._subscriptions;
+    }
+
     // Can the params here be moved to the constructor?
```
Comment (Cooryd) [inline src/rabbit-event-bus/index.ts]: Commented on the mock the same thing. Can we get rid of `register` completely in favour of constructor?

File: src/mock-event-bus/index.ts (hunk 1 above)
Comment (diversemix) [reply src/mock-event-bus/index.ts]: That's a good point I'd not thought of ... there was a reason for not using the constructor ... let me try and remember that

Comment (diversemix) [reply src/mock-event-bus/index.ts]: @Cooryd Done

File: src/rabbit-event-bus/amqp-connector.ts (hunk 2 above)
Comment (diversemix) [reply src/rabbit-event-bus/amqp-connector.ts]: @Cooryd Done

File: src/rabbit-event-bus/index.ts (hunk 3 above)
Comment (diversemix) [reply src/rabbit-event-bus/index.ts]: @Cooryd Done

**Pull request merged**